model/authentication.go

```diff
@@ -153,3 +162,22 @@ func (auth *Authentication) UpdateBy(attributes map[string]interface{}) error {
 func (auth *Authentication) Path() string {
 	return fmt.Sprintf("secret/data/%d/%s_%v_%s", auth.TenantID, auth.ResourceType, auth.ResourceID, auth.ID)
 }
+
+// mapIdExtraFields returns the ID and the Extra fields ready to be assigned to the response model, depending on
+func mapIdExtraFields(auth *Authentication) (string, map[string]interface{}) {
```

Comment:
Chewing on this a bit. I'm wondering if we should make a struct that just contains the "changeable depending on the secret store" fields. In this case it would just be an id + extra but maybe would expand in the future (for instance `version` wouldn't be present I dont think).

Reply:
I'm not sure I follow. What do you mean? Creating an embeddable struct that handles those fields?

Reply:
I'm changing my  mind the more I think about it. I was just trying to think of a better way than just passing a tuple back and forth where we would have to know that this function basically just decides on which fields are the id/extra based on the secret store.

We can change it later if we want to.

Reply:
Yep, I don't like that function either, to be honest. Feels really ugly.